src/risk_distributions/risk_distributions.py

```diff
@@ -114,328 +132,440 @@ def process(self, data: Union[np.ndarray, pd.Series], process_type: str,
         """
         return data
 
-    def pdf(self, x: pd.Series) -> Union[np.ndarray, pd.Series]:
-        ranges = self._parameter_data[['x_min', 'x_max']]
-        dist_params = self._parameter_data[self._parameter_data.columns.difference(['x_min', 'x_max'])].to_dict('series')
-        x = self.process(x, "pdf_preprocess", ranges)
-        pdf = self.distribution(**dist_params).pdf(x)
-        return self.process(pdf, "pdf_postprocess", ranges)
+    def pdf(self, x: Union[pd.Series, np.ndarray, float, int]) -> Union[pd.Series, np.ndarray, float]:
+        single_val = isinstance(x, (float, int))
+        values_only = isinstance(x, np.ndarray)
+
+        params = self.parameters.loc[:, list(self.expected_parameters)]
+        x = pd.Series(x, index=params.index)
```

Comment:
I think I'm confused about the indexing - won't this cause problems if x comes in as a series that isn't the same length/doesn't have the exact same index as params?